### PR TITLE
OAK-11105: Remove usage of Guava ImmutableMap.of() - zero/one argument case

### PR DESCRIPTION
**Pull request by reschke**

### CODE REVIEW

**Review by mbaedke — COMMENTED** [review]
Can't some of the imports already be removed, e.g. in AbstractAccessControlManagerTest?

File: oak-core-spi/src/test/java/org/apache/jackrabbit/oak/spi/whiteboard/DefaultWhiteboardTest.java [hunk 1]
```diff
@@ -38,47 +37,47 @@ public void createWhiteboard() {
...
-        whiteboard.register(Service1.class, new Service1("s1"), ImmutableMap.of());
-        whiteboard.register(Service2.class, new Service2("s2"), ImmutableMap.of("role", "myrole"));
-        whiteboard.register(Service3.class, new Service3("s3_1"), ImmutableMap.of());
-        whiteboard.register(Service3.class, new Service3("s3_2"), ImmutableMap.of("role", "myrole"));
-        whiteboard.register(Service3.class, new Service3("s3_3"), ImmutableMap.of("role", "myotherrole", "id", 1024));
+        whiteboard.register(Service1.class, new Service1("s1"), Map.of());
+        whiteboard.register(Service2.class, new Service2("s2"), Map.of("role", "myrole"));
+        whiteboard.register(Service3.class, new Service3("s3_1"), Map.of());
+        whiteboard.register(Service3.class, new Service3("s3_2"), Map.of("role", "myrole"));
+        whiteboard.register(Service3.class, new Service3("s3_3"), Map.of("role", "myotherrole", "id", 1024));
```
Comment (mbaedke) [inline oak-core-spi/src/test/java/org/apache/jackrabbit/oak/spi/whiteboard/DefaultWhiteboardTest.java]: probably not deliberate

File: oak-core-spi/src/test/java/org/apache/jackrabbit/oak/spi/whiteboard/DefaultWhiteboardTest.java [hunk 2]
```diff
@@ -38,47 +37,47 @@ public void createWhiteboard() {
...
         assertEquals(Set.of("s3_1", "s3_2", "s3_3"), track(Service3.class));
         assertEquals(Set.of("s3_1"), track(Service3.class, singletonMap("role", null)));
-        assertEquals(Set.of("s3_2"), track(Service3.class, ImmutableMap.of("role", "myrole")));
-        assertEquals(Set.of("s3_3"), track(Service3.class, ImmutableMap.of("role", "myotherrole")));
-        assertEquals(Set.of("s3_3"), track(Service3.class, ImmutableMap.of("role", "myotherrole", "id", "1024")));
-        assertEquals(Set.of("s3_3"), track(Service3.class, ImmutableMap.of("id", "1024")));
-        assertEquals(Set.of(), track(Service3.class, ImmutableMap.of("id", "2048")));
+        assertEquals(Set.of("s3_2"), track(Service3.class, Map.of("role", "myrole")));
+        assertEquals(Set.of("s3_3"), track(Service3.class, Map.of("role", "myotherrole")));
+        assertEquals(Set.of("s3_3"), track(Service3.class, Map.of("role", "myotherrole", "id", "1024")));
```
Comment (mbaedke) [inline oak-core-spi/src/test/java/org/apache/jackrabbit/oak/spi/whiteboard/DefaultWhiteboardTest.java]: dito

File: oak-core-spi/src/test/java/org/apache/jackrabbit/oak/spi/whiteboard/DefaultWhiteboardTest.java [hunk 3]
```diff
@@ -38,47 +37,47 @@ public void createWhiteboard() {
...
-        Registration r1 = whiteboard.register(Service1.class, new Service1("s1"), ImmutableMap.of());
-        Registration r2 = whiteboard.register(Service2.class, new Service2("s2"), ImmutableMap.of("role", "myrole"));
-        Registration r3_1 = whiteboard.register(Service3.class, new Service3("s3_1"), ImmutableMap.of());
-        Registration r3_2 = whiteboard.register(Service3.class, new Service3("s3_2"), ImmutableMap.of("role", "myrole"));
-        Registration r3_3 = whiteboard.register(Service3.class, new Service3("s3_3"), ImmutableMap.of("role", "myotherrole", "id", 1024));
+        Registration r1 = whiteboard.register(Service1.class, new Service1("s1"), Map.of());
+        Registration r2 = whiteboard.register(Service2.class, new Service2("s2"), Map.of("role", "myrole"));
+        Registration r3_1 = whiteboard.register(Service3.class, new Service3("s3_1"), Map.of());
+        Registration r3_2 = whiteboard.register(Service3.class, new Service3("s3_2"), Map.of("role", "myrole"));
+        Registration r3_3 = whiteboard.register(Service3.class, new Service3("s3_3"), Map.of("role", "myotherrole", "id", 1024));
```
Comment (mbaedke) [inline oak-core-spi/src/test/java/org/apache/jackrabbit/oak/spi/whiteboard/DefaultWhiteboardTest.java]: dito

File: oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexDefinition.java [hunk 4]
```diff
@@ -145,10 +145,8 @@ private Analyzer createAnalyzer() {
         if (!evaluatePathRestrictions()){
             result = defaultAnalyzer;
         } else {
-            Map<String, Analyzer> analyzerMap = ImmutableMap.<String, Analyzer>builder()
-                    .put(FieldNames.ANCESTORS,
-                            new TokenizerChain(new PathHierarchyTokenizerFactory(Collections.emptyMap())))
-                    .build();
+            Map<String, Analyzer> analyzerMap = Map.of(
+                    FieldNames.ANCESTORS, new TokenizerChain(new PathHierarchyTokenizerFactory(Collections.emptyMap())));
```
Comment (mbaedke) [inline oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexDefinition.java:149]: Deliberate?

**Comment by reschke** [comment]
Re: unneeded imports; there are a few related to the removal ImmutableSet. I'll fix those separately.

**Pull request merged**